File: app/src/main/java/com/a60n1/ejashkojme/fragment/ChatFragment.java
package com.a60n1.ejashkojme.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.a60n1.ejashkojme.R;
import com.google.android.material.tabs.TabLayout;

import java.util.Objects;

public class ChatFragment extends BaseFragment {
    private FragmentStatePagerAdapter mPagerAdapter;
    private ViewPager mViewPager;

    public ChatFragment() {
    }

    public static ChatFragment newInstance() {
        return new ChatFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_chat, container, false);

        // Create the adapter that will return a fragment for each section
        mPagerAdapter = new FragmentStatePagerAdapter(Objects.requireNonNull(getFragmentManager())) {
            private final Fragment[] mFragments = new Fragment[]{
                    new ConversationFragment(),
                    new FriendsFragment(),
            };
            private final String[] mFragmentNames = new String[]{
                    getString(R.string.heading_conversation),
                    getString(R.string.heading_friends)
            };

            @NonNull
            @Override
            public Fragment getItem(int position) {
                return mFragments[position];
            }

            @Override
            public int getCount() {
                return mFragments.length;
            }

            @Override
            public CharSequence getPageTitle(int position) {
                return mFragmentNames[position];
            }

            @Override
            public int getItemPosition(@NonNull Object object) {
                return POSITION_NONE;
            }
        };
        // Set up the ViewPager with the sections adapter.
        mViewPager = view.findViewById(R.id.chat_container);
        mViewPager.setAdapter(mPagerAdapter);
        TabLayout tabLayout = view.findViewById(R.id.chat_tabs);
        tabLayout.setupWithViewPager(mViewPager);

        return view;
    }

}

File: app/src/main/java/com/a60n1/ejashkojme/fragment/ChatFragment.kt
package com.a60n1.ejashkojme.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.a60n1.ejashkojme.R
import com.google.android.material.tabs.TabLayout

class ChatFragment : BaseFragment() {
    private var mPagerAdapter: FragmentStatePagerAdapter? = null
    private var mViewPager: ViewPager? = null
    @Suppress("RedundantOverride")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    @Suppress("DEPRECATION")
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_chat, container, false)
        // Create the adapter that will return a fragment for each section
        mPagerAdapter = object : FragmentStatePagerAdapter(fragmentManager!!) {
            private val mFragments = arrayOf<Fragment>(
                    ConversationFragment(),
                    FriendsFragment())
            private val mFragmentNames = arrayOf(
                    getString(R.string.heading_conversation),
                    getString(R.string.heading_friends)
            )

            override fun getItem(position: Int): Fragment {
                return mFragments[position]
            }

            override fun getCount(): Int {
                return mFragments.size
            }

            override fun getPageTitle(position: Int): CharSequence? {
                return mFragmentNames[position]
            }

            override fun getItemPosition(`object`: Any): Int {
                return PagerAdapter.POSITION_NONE
            }
        }
        // Set up the ViewPager with the sections adapter.
        mViewPager = view.findViewById(R.id.chat_container)
        mViewPager!!.adapter = mPagerAdapter
        val tabLayout: TabLayout = view.findViewById(R.id.chat_tabs)
        tabLayout.setupWithViewPager(mViewPager)
        return view
    }

    companion object {
        fun newInstance(): ChatFragment {
            return ChatFragment()
        }
    }
}